lib/src/health.dart

```diff
@@ -80,15 +82,28 @@ Health calcHealth({
           .map((cp) => '\n\nline ${cp.line} col ${cp.col}: ${cp.description}')
           .join();
 
-      suggestions.add(new Suggestion(
+      final suggestion = new Suggestion(
         SuggestionCode.dartanalyzerWarning,
         maxLevel,
         'Fix `$path`.',
         'Analysis of `$path` $failedWith $issueCounts$including:$issueList',
         file: path,
-      ));
+      );
+      analyzerSuggestions.add(new _Suggestion(suggestion,
+          calculateBaseHealth(errorCount, warningCount, hintCount)));
     }
   }
+  analyzerSuggestions.sort((a, b) {
+    final x = a.suggestion.compareTo(b.suggestion);
+    if (x != 0) return x;
+    return -a.percent.compareTo(b.percent);
+  });
+  var remainingScore = 100.0;
+  for (var s in analyzerSuggestions) {
+    final score = (100.0 * remainingScore * (1.0 - s.percent)).round() / 100.0;
+    remainingScore *= s.percent;
+    suggestions.add(s.suggestion.change(score: score));
```

Comment:
This seems to be the remaining score. Did you mean to add the score penalty of this suggestion, i.e. the difference between the old remainingScore and the new remainingScore?

Reply:
refactoring a bit to make it clear